form_helper documentation using article instead of post [ci skip]

notapatch opened the pull request:
### Motivation / Background

This Pull Request has been created because it will make the form_with documentation clearer.

### Detail

Documentation previously uses post which is confusing as the methods/verb is also post.

Before
`<form action="/posts" method="post">`

After
`<form action="/articles" method="post">`

### Method

Worked through a new Rails app with scaffold of article and checking the output of the HTML. Other HTML changes make it more authentic:
- /> => at the end of the input
- value attribute before name attribute

### Checklist

Before submitting the PR make sure the following are checked:

* [X] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [X] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [X] Tests are added or updated if you fix a bug or add a feature.
* [X] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

notapatch committed:
form_helper documentation using article instead of post

Documentation previously uses post which is confusing as the
methods/verb is also post.

Before
<form action="/posts" method="post">

After
<form action="/articles" method="post">

Method
Worked through a new Rails app with scaffold of article
and checking the output of the HTML. Other HTML changes
make it more authentic:
/> => at the end of the input

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -499,46 +499,44 @@ def apply_form_for_options!(object, options) # :nodoc:
-      #     <input type="text" name="post[title]" value="<the title of the post>">
+      #   <form action="/articles/1" method="post">
+      #     <input type="hidden" name="_method" value="patch" />
+      #     <input type="text" value="<the title of the article>" name="article[title]" />
```
Just noting that while the order may be rendered this way, it's sort of irrelevant (and it used to be sorted iirc)

notapatch replied:
Hi Carlos

I'll put it back in the original order then?

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -499,46 +499,44 @@ def apply_form_for_options!(object, options) # :nodoc:
+      #   <form action="/articles" method="post">
+      #     <input type="text" name="article[title]" />
       #   </form>
-      #
```
Any particular reason to removing the blank lines? We should probably keep them for better readability between examples (there's a couple more)

notapatch replied:
My mistake... I'll fix.

carlosantoniodasilva reviewed (commented):
I agree "article" will read better than "post", but we should go ahead and change it in all the examples of this form helper file at least, there's more instances. Can you take a look at those?

notapatch commented:
Hi @carlosantoniodasilva

The only references to post in the form_helper file are now related to methods.

Everything else is now article.

### Otherwise
- Added Fix you requested previously (assuming I understood properly!)
- Removed a grammatical change.
- Rebased just now.

Let me know if I can be of any assistance!

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -715,7 +714,7 @@ def apply_form_for_options!(object, options) # :nodoc:
       # In the following example the Post model has many Comments stored within it in a NoSQL database,
```
Missed one:

```suggestion
      # In the following example the Article model has many Comments stored within it in a NoSQL database,
```

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -373,7 +373,7 @@ module FormHelper
       #
```
There's a mention of `Post` above that should be updated to `Article` as well.

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -2344,65 +2343,65 @@ def fields(scope = nil, model: nil, **options, &block)
-      #   # => <label for="post_title" class="title_label">A short title</label>
+      #   # => <label for="article_title" class="title_label">A short title</label>
       #
       #   label(:privacy, "Public Post", value: "public")
```
```suggestion
      #   label(:privacy, "Public Article", value: "public")
```

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -2344,65 +2343,65 @@ def fields(scope = nil, model: nil, **options, &block)
       #
       #   label(:privacy, "Public Post", value: "public")
-      #   # => <label for="post_privacy_public">Public Post</label>
+      #   # => <label for="article_privacy_public">Public Post</label>
```
```suggestion
      #   # => <label for="article_privacy_public">Public Article</label>
```

carlosantoniodasilva commented on actionview/lib/action_view/helpers/form_helper.rb:
```diff
@@ -2563,11 +2562,11 @@ def file_field(method, options = {})
       #   <% end %>
       #
-      # In the example above, if <tt>@post</tt> is a new record, it will use "Create Post" as
+      # In the example above, if <tt>@article</tt> is a new record, it will use "Create Post" as
       # submit button label; otherwise, it uses "Update Post".
```
```suggestion
      # In the example above, if <tt>@article</tt> is a new record, it will use "Create Article" as
      # submit button label; otherwise, it uses "Update Article".
```

carlosantoniodasilva reviewed (commented):
Nice work! I found just a handful more instances that could use some updating, other than that I'll ask you to squash your commits down to 1, and we're good to merge. Thanks!

notapatch commented:
Drat! I clearly didn't search for capital p Post!

Thanks for catching that Carlos. Fixed!

Work squashed.

carlosantoniodasilva commented:
Awesome, thanks!
